### ISSUE TITLE
pubsub: pingStream should not retry to send a message.

### ISSUE THREAD

**Issue by YoshikiShibata** [googleapis/google-cloud-go#1711]
## Client
pingStream method is defined as following : https://github.com/googleapis/google-cloud-go/blob/master/pubsub/iterator.go#L472

```Go
// Send a message to the stream to keep it open. The stream will close if there's no
// traffic on it for a while. By keeping it open, we delay the start of the
// expiration timer on messages that are buffered by gRPC or elsewhere in the
// network. This matters if it takes a long time to process messages relative to the
// default ack deadline, and if the messages are small enough so that many can fit
// into the buffer.
func (it *messageIterator) pingStream() {
	// Ignore error; if the stream is broken, this doesn't matter anyway.
	_ = it.ps.Send(&pb.StreamingPullRequest{})
}
```
`Send()` will eventually call `call` method to send the empty message: https://github.com/googleapis/google-cloud-go/blob/master/pubsub/pullstream.go#L114

Inside the `call`, if `s.get` (https://github.com/googleapis/google-cloud-go/blob/master/pubsub/pullstream.go#L129) returns a closed stream, then `err = f(*spc)` fails, and then there will be a sleep. After the sleep, if `s.get` returns a closed stream again, this process will be repeated: the `call` might never return or might return much later: in this situation, `pingStream` may not return for a while.

`pingStream()` method should not retry to send the empty message, otherwise, `sender()` method (https://github.com/googleapis/google-cloud-go/blob/master/pubsub/iterator.go#L260) cannot proceed its loop in some cases.

**Comment by hongalex** [comment]
Thanks for raising this issue! Has `pingStream()` blocking become an issue for you recently? If the stream is closed (or `f(*spc)` errors for another reason), then we probably don't want to handle the rest of the items in `messageIterator.sender()`. We also [cap our sleep/backoff for 30s](https://code-review.googlesource.com/c/gocloud/+/26550/), such that if a stream is unavailable for more than 30s, we will start retrying often such that we are ready to receive messages as soon as the stream is alive again.

Let me know if there's anything I'm missing from your original description.

**Comment by hongalex** [comment]
Closing for now. Feel free to respond if you disagree with my previous comment and I'll reopen.

**Comment by YoshikiShibata** [comment]
@hongalex In production environment, I don't have any problem.

 I tried to make a fake PubSub server by implementing its gRPC because I needed the fake server to test my own service. With the fake server, I could write a test which can verify contents of  published messages.  But sometimes, clients didn't proceed its processing because of this issue I reported. Mainly because the fake server sometimes didn't keep a stream live long enough to allow the pingStream() to proceed. So it took me about three days to figure out what was wrong.
